Automattic/sensei
# Add upgrade header to Extensions page

Fixes #4753

### Changes proposed in this Pull Request

* Add a Sensei Pro featured product section in the extensions page header.

### Testing instructions

* Change the `Sensei_Extensions::SENSEILMS_PRODUCTS_API_BASE_URL` constant to `https://senseilms.mystagingwebsite.com/wp-json/senseilms-products/1.0`.
* Go to Sensei LMS -> Extensions.
* Make sure you see the "Get Sensei Pro" section.
* Make sure the CTA button points to `https://senseilms.com/checkout?add-to-cart=[product_id]`.
* Make sure the section is looking good on mobile and tablet.
* Revert the `SENSEILMS_PRODUCTS_API_BASE_URL` to its original value and make sure the `Get Sensei Pro` section is not visible and there are no errors in the console.

### New Hooks

* `sensei-lms.Extensions.hideFeaturedProduct` - Frontend filter that allows toggling the featured product.
* `sensei-lms.Extensions.featuredProduct` - Frontend filter that allows overwriting the featured product.

### Screenshot

Desktop:
<img width="1488" alt="Screenshot on 2022-02-11 at 17-41-51" src="https://user-images.githubusercontent.com/1612178/153622075-14d957b6-e94c-4016-8f56-37b03a062fd9.png">

Tablet:
<img width="911" alt="Screenshot on 2022-02-11 at 17-44-30" src="https://user-images.githubusercontent.com/1612178/153622550-5c4409b2-ecc5-4319-b139-d558b13e3ae1.png">

Mobile:
<img width="414" alt="Screenshot on 2022-02-11 at 17-45-46" src="https://user-images.githubusercontent.com/1612178/153622718-974acd78-e8f2-41cf-8a3d-8abb93497cd7.png">
